### ISSUE TITLE
szuru_sql exits, PostgreSQL appears to contain a database

### ISSUE THREAD

**Issue by ziimro** [rr-/szurubooru#393]
i followed this video = [https://www.youtube.com/watch?v=V73M_COXa5k&t=237s](url)

My port in .env is 4040
I didint change the domain in config.yaml and didint change base_url=/ in .env

```
server_1  | standard_init_linux.go:207: exec user process caused "exec format error"
client_1  | standard_init_linux.go:207: exec user process caused "exec format error"
sql_1     |
sql_1     | PostgreSQL Database directory appears to contain a database; Skipping initialization
sql_1     |
sql_1     | 1970-04-27 10:17:12.009 GMT [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
sql_1     | 1970-04-27 10:17:12.009 GMT [1] LOG:  listening on IPv6 address "::", port 5432
sql_1     | 1970-04-27 10:17:12.009 GMT [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
sql_1     | 1970-04-27 10:17:12.009 GMT [1] LOG:  startup process (PID 20) was terminated by signal 11: Segmentation fault
sql_1     | 1970-04-27 10:17:12.009 GMT [1] LOG:  aborting startup due to startup process failure
sql_1     | 1970-04-27 10:17:12.009 GMT [1] LOG:  database system is shut down
sql_1     |
sql_1     | PostgreSQL Database directory appears to contain a database; Skipping initialization
szuru_client_1 exited with code 1
szuru_server_1 exited with code 1
```

and then like couple lines of this
```
sql_1     | PostgreSQL Database directory appears to contain a database; Skipping initialization
sql_1     |
sql_1     | 1970-04-27 10:17:12.009 GMT [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
sql_1     | 1970-04-27 10:17:12.009 GMT [1] LOG:  listening on IPv6 address "::", port 5432
sql_1     | 1970-04-27 10:17:12.009 GMT [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
sql_1     | 1970-04-27 10:17:12.009 GMT [1] LOG:  startup process (PID 20) was terminated by signal 11: Segmentation fault
sql_1     | 1970-04-27 10:17:12.009 GMT [1] LOG:  aborting startup due to startup process failure
sql_1     | 1970-04-27 10:17:12.009 GMT [1] LOG:  database system is shut down
szuru_sql_1 exited with code 1
```

**Comment by sgsunder** [comment]
The message `PostgreSQL Database directory appears to contain a database; Skipping initialization` is normal (Basically there's a startup script that creates a new database if it's the first time you started up the postgres container).

The real issue is from the segmentation fault, and the "exec format error." What platform are you running on? The provided containers are compiled for x86-64 only, if you are running on something else then you would have to build the containers yourself.

**Comment by ziimro** [comment]
im running that on raspberry pi 3 B+ with the standard raspberry pi debian os
i also have postgres, nextcloud and bitwarden running on that, maybe thats the issue

**Comment by sgsunder** [comment]
See here for ARM instructions: https://github.com/rr-/szurubooru/wiki/ARM-and-Raspberry-Pi-Support-for-Docker-builds

**Comment by ziimro** [comment]
Now i got this
` WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.13/main: temporary error (try again later)
WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.13/community: temporary error (try again later)
ERROR: unable to select packages:
  dumb-init (no such package):
    required by: world[dumb-init]
ERROR: Service 'client' failed to build: The command '/bin/sh -c apk --no-cache add dumb-init' returned a non-zero code: 1`

**Comment by sgsunder** [comment]
> try again later

Did you try again to see if it still fails? It seems like the package repo might have been down for a bit

**Comment by ziimro** [comment]
nope, still the same issue

**Comment by sgsunder** [comment]
What Dockerfile are you building? We're still on Alpine Linux v3.12 inside the Docker containers, so that error message doesn't make sense.

**Comment by ziimro** [comment]
`
## Example Docker Compose configuration
##
## Use this as a template to set up docker-compose, or as guide to set up other
## orchestration services
version: '2'

services:

  server:
    build:
      context: ./server
    depends_on:
      - sql
    environment:
      ## These should be the names of the dependent containers listed below,
      ## or FQDNs/IP addresses if these services are running outside of Docker
      POSTGRES_HOST: sql
      ## Credentials for database:
      POSTGRES_USER:
      POSTGRES_PASSWORD:
      ## Commented Values are Default:
      #POSTGRES_DB: defaults to same as POSTGRES_USER
      #POSTGRES_PORT: 5432
      #LOG_SQL: 0 (1 for verbose SQL logs)
    volumes:
      - "${MOUNT_DATA}:/data"
      - "./server/config.yaml:/opt/app/config.yaml"

  client:
    build:
      context: ./client
    depends_on:
      - server
    environment:
      BACKEND_HOST: server
      BASE_URL:
    volumes:
      - "${MOUNT_DATA}:/data:ro"
    ports:
      - "${PORT}:80"

  sql:
    image: postgres:11-alpine
    restart: unless-stopped
    environment:
      POSTGRES_USER:
      POSTGRES_PASSWORD:
    volumes:
      - "${MOUNT_SQL}:/var/lib/postgresql/data"
`

**Comment by ziimro** [comment]
My docker file in server/Dockerfile shows `ALPINE_VERSION=3.12`

**Comment by sgsunder** [comment]
I can't reproduce it at all no matter what I try, even in ARM emulation. Perhaps it's an internet issue? Your DNS resolver is borked or something.

Does running `nslookup dl-cdn.alpinelinux.org` work? How about `docker run --rm alpine:3.12 nslookup dl-cdn.alpinelinux.org`?

**Comment by ziimro** [comment]
i ran `docker run --rm alpine:3.12 nslookup dl-cdn.alpinelinux.org`
```
nslookup: read: Connection refused
nslookup: read: Connection refused
nslookup: read: Connection refused
nslookup: read: Connection refused
Server:         8.8.4.4
Address:        8.8.4.4:53

Non-authoritative answer:
dl-cdn.alpinelinux.org  canonical name = dualstack.d.sni.global.fastly.net
Name:   dualstack.d.sni.global.fastly.net
Address: 2a04:4e42::645
Name:   dualstack.d.sni.global.fastly.net
Address: 2a04:4e42:200::645
Name:   dualstack.d.sni.global.fastly.net
Address: 2a04:4e42:400::645
Name:   dualstack.d.sni.global.fastly.net
Address: 2a04:4e42:600::645

Non-authoritative answer:
dl-cdn.alpinelinux.org  canonical name = dualstack.d.sni.global.fastly.net
Name:   dualstack.d.sni.global.fastly.net
Address: 151.101.2.133
Name:   dualstack.d.sni.global.fastly.net
Address: 151.101.66.133
Name:   dualstack.d.sni.global.fastly.net
Address: 151.101.130.133
Name:   dualstack.d.sni.global.fastly.net
Address: 151.101.194.133
```

**Comment by sgsunder** [comment]
> `nslookup: read: Connection refused`

I think your Docker install is not resolving DNS properly. Don't know how to solve that for sure, but you can try restarting the Docker app, or reinstalling it

**Comment by sgsunder** [comment]
@ziimro ARM images are now available on Docker Hub! check the wiki for further details. I'm not sure if you're still having this issue but hopefully this makes the build easier for you.

Feel free to reopen if this issue is still unresolved.

**Comment by ziimro** [comment]
@sgsunder is been some time and i tried tinkering with it again. I changed to alpine 13.3 in docker compose but it still doesent work. Id love to get some help and finally get it working

**Comment by ziimro** [comment]
` PostgreSQL Database directory appears to contain a database; Skipping initialization
sql_1     |
sql_1     | 1970-04-28 06:32:08.010 GMT [1] LOG:  starting PostgreSQL 13.3 on arm-unknown-linux-musleabihf, compiled by gcc (Alpine 10.3.1_git20210424) 10.3.1 20210424, 32-bit
sql_1     | 1970-04-28 06:32:08.010 GMT [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
sql_1     | 1970-04-28 06:32:08.010 GMT [1] LOG:  listening on IPv6 address "::", port 5432
sql_1     | 1970-04-28 06:32:08.010 GMT [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
sql_1     | 1970-04-28 06:32:08.010 GMT [1] LOG:  startup process (PID 20) was terminated by signal 11: Segmentation fault
sql_1     | 1970-04-28 06:32:08.010 GMT [1] LOG:  aborting startup due to startup process failure
sql_1     | 1970-04-28 06:32:08.010 GMT [1] LOG:  database system is shut down
szuru_sql_1 exited with code 1`
same issues over and over again

**Comment by ziimro** [comment]
i tested it on vms and it works just fine on ubuntu server (i have the classic raspi os on my pi)

**Comment by sgsunder** [comment]
Can you post the result of `docker info | grep -i architecture`?

**Comment by ziimro** [comment]
> 

`errors pretty printing info`

**Comment by ziimro** [comment]
@sgsunder sorry wrong machine
```
WARNING: No memory limit support
WARNING: No swap limit support
WARNING: No kernel memory limit support
WARNING: No oom kill disable support
Architecture: armv7l
```
maybe its an issue with me running on armv7 and not having updated versions of dependencies

**Comment by ziimro** [comment]
i have a rasberry pi 4 on shipping so i will test it there too once it arrives

**Comment by squeevee** [comment]
I've gotten very similar errors to these trying to get szurubooru to work on a raspberry pi 4 running debian. I've now got it working, with two major tweaks:

I had to edit the Dockerfile for client and server, change every instance of `$BUILDPLATFORM` to `linux/arm/v7` and NOT have `DOCKER_BUILDKIT` in my environment when running `docker-compose build`

It only seems to work using the Docker Engine package provided by Moby rather than the one in the Debian distro, see https://docs.docker.com/engine/install/debian/

The "temporary problem" error message in `apk` seems to be related to a change to [Alpine 3.13][1] which required special workarounds to work with older versions of Docker and libseccomp on the host system. At the time Alpine 3.13 was released, Debian's packages had not met the new version requirements; they now do, but for some reason if BuildKit is used I get that error again (seemingly regardless of which package I use).

[1]: https://wiki.alpinelinux.org/wiki/Release_Notes_for_Alpine_3.13.0#musl_1.2

I don't have a specific guess why the Debian package for Docker Engine doesn't work, except that the Moby package has higher versions of containerd and runc. I'm pretty sure I was using the Debian package six months ago and running szurubooru without major problems, but I've wiped this pi since then so who's to say.
